pkg/client/rules.go

```diff
@@ -32,18 +47,31 @@ func (r *CortexClient) CreateRuleGroup(ctx context.Context, namespace string, rg
 
 // DeleteRuleGroup creates a new rule group
 func (r *CortexClient) DeleteRuleGroup(ctx context.Context, namespace, groupName string) error {
+	path := rulerAPIPath
+	if r.legacy {
+		path = legacyAPIPath
+	}
 	escapedNamespace := url.PathEscape(namespace)
 	escapedGroupName := url.PathEscape(groupName)
+	path += "/" + escapedNamespace + "/" + escapedGroupName
 
-	_, err := r.doRequest("/api/prom/rules/"+escapedNamespace+"/"+escapedGroupName, "DELETE", nil)
+	log.WithFields(log.Fields{
+		"url": path,
+	}).Debugln("path built to request rule group")
+
+	_, err := r.doRequest(path, "DELETE", nil)
 	return err
 }
 
 // GetRuleGroup retrieves a rule group
 func (r *CortexClient) GetRuleGroup(ctx context.Context, namespace, groupName string) (*rwrulefmt.RuleGroup, error) {
+	path := rulerAPIPath
+	if r.legacy {
+		path = legacyAPIPath
+	}
 	escapedNamespace := url.PathEscape(namespace)
 	escapedGroupName := url.PathEscape(groupName)
-	path := "/api/prom/rules/" + escapedNamespace + "/" + escapedGroupName
+	path += "/" + escapedNamespace + "/" + escapedGroupName
```

Comment:
This can be simplified if the path prefix is stored as a string in the client struct.

Reply:
I agree. I'll simplify it.